napari/utils/events/evented_model.py

```diff
@@ -303,41 +366,16 @@ def __setattr__(self, name: str, value: Any) -> None:
             dep for dep, has_cb in has_callbacks.items() if has_cb
         ]
 
-        before = getattr(self, name, object())
-        before_deps = {}
-        with warnings.catch_warnings():
-            # we still need to check for deprecated properties
-            warnings.simplefilter("ignore", DeprecationWarning)
-            for dep in dep_with_callbacks:
-                before_deps[dep] = getattr(self, dep, object())
+        if name not in self._check_queue:
```

Comment:
I guess this `if` check is because you could potentially have:
- `a` is added to the queue, with value `0`
- `a` changes somehow to `1`
- `a` is added to the queue again, with new value `1`
- `a` is checked against `1`, no event is emitted

Maybe impossible? Still, good idea to avoid it :P

Reply:
Exactly this scenario. And strictly possible when you think about properties, especially aliased because of deprecations.